### PR TITLE
add faust-layer, remove faust from extra-langs

### PR DESCRIPTION
**Pull request by magnetophon**
My first spacemacs layer, total elisp newb.
Does what it says in the README, with one bug:

I have ```auto-completion-enable-snippets-in-popup t``` in my .spacemacs
In faust files, only yas-keys that have a similar word in an open buffer
show up in the company menu.

Example: With no open buffers ```ne``` doesn't show a company menu.
If a file has the word ```nent``` in it, I get a menu with ```ne -> nentry``` (the snippet) and ```nent```.

Having ```ne$AnyTwoCharacters``` in the file also works, but just ```ne``` or ```nen``` doesn't.

PS: The snippets are [here](https://github.com/AndreaCrotti/yasnippet-snippets/tree/master/faust-mode), so they'll automatically end up in spacemacs after a while, right?

### CODE REVIEW

File: layers/+lang/faust/packages.el [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        installed using the Emacs package manager.
+
+      - The symbol `local' directs Spacemacs to load the file at
+        `./local/PACKAGE/PACKAGE.el'
+
+      - A list beginning with the symbol `recipe' is a melpa
+        recipe.  See: https://github.com/milkypostman/melpa#recipe-format")
+
+(defun faust/init-faust-mode ()
+  (use-package faust-mode :defer t :mode "\\.\\(dsp\\|lib\\)\\'"))
```
Comment (person808) [inline layers/+lang/faust/packages.el]: Dont put this all on one line.

```elisp
(use-package faust-mode
  :defer t
  :mode "\\.\\(dsp\\|lib\\)\\'")
```

**Comment by magnetophon** [comment]
@person808 OK, done.

**Comment by robbyoconnor** [comment]
Why is this a layer?

**Comment by magnetophon** [comment]
@robbyoconnor I was looking trough the other extra-langs and they sm to be mostly syntax HL.
I also saw some separate layers that are as simpe as this one, so I just mimicked.

If you want I can put it back in extra-langs.

**Comment by robbyoconnor** [comment]
I'm just a dude who cares about the project -- by all means keep it as-is. @syl20bnr has final say. The rest of us just care a lot.

**Comment by StreakyCobra** [comment]
@magnetophon What you are doing here is fine. Wen there are some extra configuration for a package it's better to have it in its own layer. There were also some criticism lately about this extra-lang layer, so I'm fine if we start moving some on their own layer :+1:

**Comment by robbyoconnor** [comment]
I was in no way saying it shouldn't be this way...jus asking why :)

File: layers/+lang/faust/packages.el [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+;; Briefly, each package to be installed or configured by this layer should be
+;; added to `faust-packages'. Then, for each package PACKAGE:
+;;
+;; - If PACKAGE is not referenced by any other Spacemacs layer, define a
+;;   function `faust/init-PACKAGE' to load and initialize the package.
+
+;; - Otherwise, PACKAGE is already referenced by another Spacemacs layer, so
+;;   define the functions `faust/pre-init-PACKAGE' and/or
+;;   `faust/post-init-PACKAGE' to customize the package as it is loaded.
+
```
Comment (robbyoconnor) [inline layers/+lang/faust/packages.el]: Kill these comments -- they're not needed. They're instructional

File: layers/+lang/faust/packages.el [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      - The symbol `elpa' (default) means PACKAGE will be
+        installed using the Emacs package manager.
+
+      - The symbol `local' directs Spacemacs to load the file at
+        `./local/PACKAGE/PACKAGE.el'
+
+      - A list beginning with the symbol `recipe' is a melpa
+        recipe.  See: https://github.com/milkypostman/melpa#recipe-format")
+
```
Comment (robbyoconnor) [inline layers/+lang/faust/packages.el]: Same here -- kill the comments unless needed.

File: layers/+lang/faust/packages.el [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+;;   define the functions `faust/pre-init-PACKAGE' and/or
+;;   `faust/post-init-PACKAGE' to customize the package as it is loaded.
+
+;;; Code:
+
+(defconst faust-packages
+  '(faust-mode
+    company
+    yasnippet
+    )
```
Comment (robbyoconnor) [inline layers/+lang/faust/packages.el]: Change this to (with the closing paren on the last line. 
```emacs-lisp
(defconst faust-packages
  '(faust-mode
    company
    yasnippet)
```

Comment (AlejandroCatalina) [inline layers/+lang/faust/packages.el]: Actually also alphabetical order is required

Comment (robbyoconnor) [inline layers/+lang/faust/packages.el]: That too. fix it :)

**Comment by magnetophon** [comment]
@robbyoconnor OK, done.

**Comment by magnetophon** [comment]
Any chance of this being merged?
Any changes you want me to make?

**Comment by robbyoconnor** [comment]
It tends to take time to get things merged. Hold tight.

**Comment by magnetophon** [comment]
Ok, thanks!

**Comment by robbyoconnor** [comment]
Also squash your commits please.

**Comment by robbyoconnor** [comment]
`SPC g r I`

**Comment by robbyoconnor** [comment]
`SPC g r i`

**Comment by syl20bnr** [comment]
Thank you!
I refactored it and changed the key binding to make them meet the conventions, related commit https://github.com/syl20bnr/spacemacs/commit/8e6fd86
Cherry-picked into develop branch, you can safely delete your branch.

**Comment by magnetophon** [comment]
@syl20bnr Great, thanks!

**Comment by robbyoconnor** [comment]
Hi,

I'd like to add you to my professional network on LinkedIn.

- Robert

Confirm that you know Robert: https://www.linkedin.com/e/v2?e=0-ini67odh-6g&t=ssuw&tracking=eml-guest-invite-cta&ek=invite_guest&sharedKey=ErrBrsq2&invitationId=6130896966860763136








You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=0-ini67odh-6g&t=lun&midToken=AQFmrFVhGEd36Q&ek=invite_guest&loid=AQEchUgqN0eRSwAAAVRVSclXl-Oaan8jl5pzKrGgJ4znICMsHUVbZQUf3yZ7HYyOP0RJhot9vWo2u6Ww9Yy8lnCc0oOGVgxH1a9Q5cjovbtxCwqXl24Q59D1qHUTUjfegTIgJ7IXsC4VjF9VAUB6hmEEIyxKrIS9MtPb1Vx2TCvl5Xnd9J-fXtyFMn2JLuxhLwjI&eid=0-ini67odh-6g


This email was sent to reply@reply.github.com.

If you need assistance or have questions, please contact LinkedIn Customer Service: https://www.linkedin.com/e/v2?e=0-ini67odh-6g&a=customerServiceUrl&ek=invite_guest

&copy; 2016 LinkedIn Corporation, 2029 Stierlin Court, Mountain View CA 94043. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.

**Pull request closed without merging**